src/data.ml

```diff
@@ -114,11 +108,38 @@ end = struct
 
   let hash_size = 30
 
-  let encode s = s
+  let encode s f = f s
 
   let decode s off = String.sub s off L.length
 
   let encoded_size = L.length
 
   let equal = String.equal
 end
+
+module Double_String_fixed (L : sig
+  val length : int
+end) : sig
+  type t = string * string
+
+  include Key with type t := string * string
+
+  include Value with type t := string * string
+end = struct
+  type t = string * string [@@deriving repr]
```

Comment:
We can't delete those since they're later used in the functor and use in index.ml, for example, to do this:

```ocaml
let pp_key = Repr.pp K.t
```